### DISCUSSION TITLE
Drawer host animation flickering

### DISCUSSION THREAD

**Discussion by abogadi** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3253] [Q&A] [answered]
Hi All!

Quite new to use Material design in Xaml.  
I have a weird issue whereby the drawer host animation flickers and stutters in my wpf app.
I copied the template from the demo as is, just removed the drawers I did not need.  BTW the drawer works fine on the demo app.
Any ideas?

https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/62957678/a7f90fa8-24f6-4252-b692-556ced691ad5

**Comment by Keboo** [comment, marked as answer]
My first thought is there is some sort of binding that is doing work/changing values during the animation. It is also possible that the size of the content within the drawer is changing as part of the animation causing it. 

Two things I would check:
1. Set the root content in the drawer to a fixed width and see if that changes the behavior. If it does that means the issue the content size changing as part of the animation.
2. If you are opening the drawer by binding to its is-open property, check the contents of the getter/setter of the bound property to make sure they are not causing any recursive calls/work as part of the trigger.

**Reply by abogadi** [reply]
It was the width.
Many thanks for the rapid help!
